scipy/spatial/_geometric_slerp.py

```diff
@@ -28,10 +32,12 @@ def _geometric_slerp(start, end, t):
     return start * c[:, np.newaxis] + end * s[:, np.newaxis]
 
 
-def geometric_slerp(start,
-                    end,
-                    t,
-                    tol=1e-7):
+def geometric_slerp(
+    start: npt.ArrayLike,
+    end: npt.ArrayLike,
+    t: npt.ArrayLike,
+    tol: float = 1e-7,
```

Comment:
Note that an explicit `isinstance(t, float)` check is performed by the function, 
so we can't use [`DecimalNumber`](https://github.com/scipy/scipy/blob/2c72ae27e59e64f47d64bf65800a291ff972df86/scipy/_lib/_util.py#L20) here no matter if we'd want to.